controlplane/kubeadm/controllers/controller.go

```diff
@@ -422,6 +435,45 @@ func (r *KubeadmControlPlaneReconciler) reconcileHealth(ctx context.Context, clu
 	return nil
 }
 
+// ReconcileInfrastructureMachineRefs adds infrastructure template ref as annotation to the infrastructure objects missing it.
+func (r *KubeadmControlPlaneReconciler) ReconcileInfrastructureMachineRefs(ctx context.Context, infraTemplateRef corev1.ObjectReference, machines internal.FilterableMachineCollection) (bool, error) {
+	isPatched := false
+	for _, machine := range machines {
+		if !machine.DeletionTimestamp.IsZero() {
+			// Machines to be deleted will be handled by the reconcileHealth logic.
+			continue
+		}
+
+		infraObj, err := external.Get(ctx, r.Client, &machine.Spec.InfrastructureRef, machine.Spec.InfrastructureRef.Namespace)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return isPatched, errors.Wrapf(err, "machine's infrastructure object is missing")
+			}
+			return isPatched, errors.Wrapf(err, "failed to get Machine's infrastructure object")
```

Comment:
Since we're not doing anything different (except wrapping with a different message), I don't think we need to handle these errors differently. Also was using `Wrapf` without any format flags in the string.

```suggestion
			return isPatched, errors.Wrap(err, "failed to get Machine's infrastructure object")
```